src/sinks/kafka.rs

```diff
@@ -392,6 +405,28 @@ impl Sink<Event> for KafkaSink {
     }
 }
 
+fn get_headers(event: &Event, headers_key: &str) -> Option<OwnedHeaders> {
+    if let Event::Log(log) = event {
+        if let Some(headers) = log.get(headers_key) {
+            match headers {
+                Value::Map(headers_map) => {
+                    let mut owned_headers = OwnedHeaders::new_with_capacity(headers_map.len());
+                    for (key, value) in headers_map {
+                        if let Value::Bytes(value_bytes) = value {
+                            owned_headers = owned_headers.add(key, value_bytes.as_ref());
+                        } else {
+                            warn!("Header value for key {} must be a UTF-8 encoded String (Bytes). Ignoring.", key);
+                        }
+                    }
+                    return Some(owned_headers);
+                }
+                _ => warn!("Value of headers_key field must be a map. Ignoring."),
```

Comment:
I think this should be moved into an internal event, possibly incrementing an error counter.

Reply:
Yeah, agreed, same with the warning above this one. You can see Bruce's recent PR adding more events to https://github.com/vectordotdev/vector/pull/9205 for an example of error internal events.